### PR TITLE
create default config with `prefect make user config`

### LINKED ISSUE #904: make user config should create default
User configurability may be unclear to the user, so I think `prefect make-user-config` should give the user more guidance about their options than an empty file

I propose that it should copy the defaults to the user config location

However, I could just as easily see doing both of the following, instead of this PR--
1) adding a note to the `prefect make-user-config` part of the docs (docs/guide/core_concepts) about where to find the default file 
2) adding "the default configuration is located at src/prefect/config.toml" to the auto-generated settings file

### PR DESCRIPTION
**Pull request by TylerWanner**
closes #904 
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [ ] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

## What does this PR change?
`prefect make-user-config` generate the default file instead of just comments

## Why is this PR important?
Provides the user their full workbench of settings to manipulate

### CODE REVIEW

**Comment by cicdw** [comment]
@TylerWanner you'll need to run `black .` from the base of your repository to format the code properly

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ These changes are available in the [master branch](https://github.com/PrefectHQ/
 
 ### Enhancements
 
-- None
+- `prefect make-user-config` now creates default user config from src/prefect/config.toml
```
Comment (cicdw) [inline CHANGELOG.md]:
```suggestion
- `prefect make-user-config` now creates default user config from src/prefect/config.toml - [#904](https://github.com/PrefectHQ/prefect/issues/904)
```
we typically link the change to the issue it closes; if no issue exists, we link directly to the PR

File: src/prefect/cli/__init__.py [hunk 2]
```diff
@@ -33,8 +33,12 @@ def make_user_config():
     with open(user_config_path, "w") as user_config:
         user_config.write(
             "# This is a user configuration file.\n"
-            "# Settings placed here will overwrite Prefect's defaults."
+            "# Settings placed here will overwrite Prefect's defaults.\n"
         )
+        config_loc = os.path.join(__file__, "../config.toml")
+        with open(config_loc) as defaults_file:
```
Comment (cicdw) [inline src/prefect/cli/__init__.py]:
```suggestion
        with open(config_loc, "r") as defaults_file:
```
While `"r"` is the default mode for open, it's standard practice to include it explicitly.

File: src/prefect/cli/__init__.py [hunk 3]
```diff
@@ -33,8 +33,12 @@ def make_user_config():
...
+            for line in defaults_file:
+                user_config.write()
```
Comment (cicdw) [inline src/prefect/cli/__init__.py]: So this writes a blank line; you probably want either:
```python
user_config.write(defaults_file.read())
```
_or_
```python
for line in defaults_file.readlines():
    user_config.write(line)
```
it's basically personal preference as to which one you choose.

**Comment by jlowin** [comment]
On further thought, we shouldn’t pollute the user config even with comments, as it will make portability more difficult in the future. We can achieve the same purpose in a much more accessible way by simply publishing the default config in the API reference.

**Comment by cicdw** [comment]
I’m all for publishing the config in the API docs, but not providing a simple mechanism for generating a template seems overly strict to me. I’ll admit I just copied our default config to my user config location a long time ago, and just update it as new things come along / my settings stop working.

I don’t feel strongly either way, but if we really aren’t going to implement this, then we should repurpose this PR to remove the make user config CLI.

**Comment by jlowin** [comment]
Generating a separate template config is an interesting idea, though I'm not sure it solves any use cases that proper documentation doesn't also solve in a way that looks better, is more discoverable, and can be searched. I'm left wondering what proportion of users are generating configs and *then* looking for documentation, rather than the other way around? It seems far more likely that a user would read about a config somewhere in the docs, and then go set it, then blindly run a minor CLI command. 

I'd consider generating a template as long as it was separate from the user config itself. The reason is: if you generated that template config today, and edited it to enable some config setting, then you'd be prohibited from running it again in the future because doing so would overwrite your config. However, if the template is just a separate file that gets generated from the default config, I'm back to wondering under what situation is that a UX improvement over simply adding it to the docs for all users, and not requiring a command to be run at all to get an up-to-date reference?

Lastly I don't see the argument for removing the CLI command, because 1) user configs are important and 2) without a CLI command users have to load Prefect, query the user config location, and create an appropriately-named file at that location. I doubt there's any user for whom that would be a preferable experience to running a simple CLI command.

**Comment by cicdw** [comment]
> The reason is: if you generated that template config today, and edited it to enable some config setting, then you'd be prohibited from running it again in the future because doing so would overwrite your config. 

This argument applies equally to having a static user config file at all.

>  I'm back to wondering under what situation is that a UX improvement over simply adding it to the docs for all users, and not requiring a command to be run at all to get an up-to-date reference?

Do you really not see a UX benefit to running `prefect make-user-config`, opening the file, editing the one setting you knew was in there but couldn't quite remember where / what it was called VS. visiting the docs website, trying to remember that one place the config was listed and copy / pasting?

> Lastly I don't see the argument for removing the CLI command, because 1) user configs are important and 2) without a CLI command users have to load Prefect, query the user config location, and create an appropriately-named file at that location.

The command literally doesn't do anything; to use your own argument, I don't see what this command solves that isn't solved by proper documentation of where a user should place their config file (the only piece of information the command produces is a file location).

To be clear, I don't have a strong opinion either way but I _do_ have a strong opinion on pursuing one of these two options:
- either the `make-user-config` CLI is maximally useful in a local setting and provides enough information that docs aren't required to get a full user config up and running
- we remove `make-user-config` and rely on good documentation

**Comment by cicdw** [comment]
After some further thought, I _do_ agree there is a potential issue with defaults persisting after the first creation of the file, so I'm all in on removing this CLI endpoint.

**Comment by joshmeek** [comment]
I agree, removing this CLI endpoint will both alleviate that default problem that could happen without user knowledge and it will avoid future confusion about what this command is.

**Comment by TylerWanner** [comment]
Closing this PR.  Will submit a new PR for deleting the CLI endpoint

**Pull request closed without merging**